core/server/src/main/java/alluxio/worker/block/BlockMasterSync.java

```diff
@@ -205,7 +210,7 @@ private void handleMasterCommand(Command cmd) throws Exception {
         break;
       // Master requests re-registration
       case Register:
-        WorkerIdRegistry.registerWithBlockMaster(mMasterClient, mWorkerAddress);
+        mWorkerId.set(mMasterClient.getId(mWorkerAddress));
```

Comment:
It would be nice to only have the read only version, and have the mutable version only in one class. Given we can't achieve that here I think the read only version would be more confusing than not since its not clear why some classes can modify it and others cannot.